odoo/tools/config.py

```diff
@@ -619,7 +621,7 @@ def _postprocess_options(self):
         ]))
 
         self._runtime_options['init'] = dict.fromkeys(self['init'], True) or {}
-        self._runtime_options['demo'] = dict(self['init']) if not self['without_demo'] else {}
+        self._runtime_options['demo'] = dict(self['init']) if self['with_demo'] else {}
```

Comment:
You could check both flags until without-demo is removed.
```suggestion
        self._runtime_options['demo'] = dict(self['init']) if self['with_demo'] or not self['without_demo'] else {}
```